files/build_templates/docker_image_ctl.j2

```diff
@@ -418,13 +418,28 @@ start() {
     {%- if docker_container_name == "swss" %}
     # Generate the asic_table.json and peripheral_table.json
     if [ ! -f /etc/sonic/asic_table.json ] && [ -f /usr/share/sonic/templates/asic_table.j2 ]; then
-        sonic-cfggen -d -t /usr/share/sonic/templates/asic_table.j2 > /etc/sonic/asic_table.json
+        sonic-cfggen -d -t /usr/share/sonic/templates/asic_table.j2 > /etc/sonic/asic_table.json 2> errorlog.txt
+        if [[ $? -ne 0 ]] ; then
+            echo "sonic-cfggen failed to render asic_table.json, try working around it"
+            echo "$(cat errorlog.txt)"
+            sonic-cfggen -a '{"DEVICE_METADATA":{"localhost":{"platform":"'$PLATFORM'"}}}' -t /usr/share/sonic/templates/asic_table.j2 > /etc/sonic/asic_table.json
```

Comment:
Use -a can work around, but there is a potential issue: if these templates update in future to use some data in other CONFIG_DB tables, and who change it doesn't know this work around. then when this issue happens, the rendered file will break.

Is there any solution to prevent that happen?

Reply:
Theoretically, you are correct. But this is a very short-term WR.
During this time, we do not expect any change in asic_table.json.